pkgs/by-name/fl/flye/package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, zlib
+, curl
+, libdeflate
+, bash
+, coreutils
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "flye";
+  version = "2.9.3";
+
+  src = fetchFromGitHub {
+    owner = "fenderglass";
+    repo = "flye";
+    rev = version;
+    hash = "sha256-IALqtIPmvDYoH4w/tk2WB/P/pAcKXxgnsu9PFp+wIes=";
+  };
+
+  nativeCheckInputs = [ python3Packages.pytestCheckHook ];
+
+  propagatedBuildInputs = [ coreutils ];
+
+  buildInputs = [
+    zlib
+    curl
+    libdeflate
+  ];
+
+  patches = [ ./aarch64-fix.patch ];
```

Comment:
no need to change, but you could make patch optional just for aarch64-linux

Reply:
I considered that but since the patch only affects `aarch64` I figured simpler would be better. Is making it optional a better practice?

Reply:
it would  depend on what upsteam accepts,

either way it's an if statement, it can be in Makefile or nix

Reply:
I will mark this as resolved. I left it open while I considered the change, but I've concluded that patching the source indiscriminately seems better to me, as it is more consistent with how it handles `aarch64-darwin` and mirrors the fix I've proposed to upstream.

Unless I misunderstood your messages you appear to not prefer one way over the other which tells me one is not more idiomatic than the other. If that is the case please let me know, as that would affect my decision.